Next (#2904)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/about-cloud/architecture.md

---
title: "dbt Cloud Architecture"
id: "architecture"
description: "Information about the architecture, communication, and security of dbt Cloud"
---

> This page is intended to help practitioners and those interested in the architecture and data flow of the hosted dbt Cloud product.

Reviewer: ```suggestion
This page helps practitioners and those interested in dbt Cloud's architecture and data flow.
```

## About dbt Cloud architecture

> The dbt Cloud application comprises a set of static and dynamic components. The static components are constantly running to serve highly available dbt Cloud functionality, for example, the dbt Cloud web application. The dynamic components are created ad-hoc to fill background jobs or requests to use the IDE. 

Reviewer: ```suggestion
The dbt Cloud application has two types of components: static and dynamic. The static components are always running to serve highly available dbt Cloud functions, like the dbt Cloud web application. On the other hand, the dynamic components are created ad-hoc to handle tasks such as background jobs or requests to use the IDE.
```

> dbt Cloud is available in most regions around the world in both [single tenant](/docs/cloud/about-cloud/tenancy#single-tenant)(AWS and Azure) and [multi-tenant](/docs/cloud/about-cloud/tenancy#multi-tenant) configurations.  

Reviewer: ```suggestion
dbt Cloud is available in most regions around the world in both [single tenant](/docs/cloud/about-cloud/tenancy#single-tenant) (AWS and Azure) and [multi-tenant](/docs/cloud/about-cloud/tenancy#multi-tenant) configurations.  
```

dbt Cloud uses PostgreSQL for its backend, S3-compatible Object Storage systems for logs and artifacts, and a Kubernetes storage solution for creating dynamic, persistent volumes. 

> All data at rest on dbt Cloud servers is protected using AES-256 encryption. 

Reviewer: just double checking if it's meant to say 'at rest'?

Reviewer 2: Yup! When the data is on our servers (vs in transit over the internet), it is considered "at rest." It's a different type of encryption than when the data is moving over the internet

<img src="/img/docs/dbt-cloud/on-premises/data-flows.png" />

> For a more detailed breakdown of the dbt Cloud apps, [download the advanced architecture guide PDF](https://drive.google.com/uc?export=download&id=1lktNuMZybXfqFtr24J8zAssEfoL9r51S).

Reviewer: ```suggestion
For a more detailed breakdown of the dbt Cloud application, [download the advanced architecture guide PDF](https://drive.google.com/uc?export=download&id=1lktNuMZybXfqFtr24J8zAssEfoL9r51S).
```

Reviewer 2: This should say "apps" plural. While the Cloud product can be called an application this is referencing individual apps all doing different things on their own and together behind the scenes.

## Communication

dbt Cloud can communicate with several external services, including data platforms, git repositories, authentication services, and directories. All communications occur over HTTPS (attempts to connect via HTTP are redirected to HTTPS). dbt Cloud encrypts in transit using the TLS 1.2 cryptographic protocol. 

> TLS (Transport Layer Security) 1.2 is an industry-standard protocol for encrypting sensitive data while it travels over the public internet (which does not offer native encryption).

Reviewer: love this explanation

> A typical scenario that might be seen frequently is an employee working in a public space, such as an airport or café. The user might be connected to an unsecured public network offered by a facility to which many others are also connected. What if there is a bad actor amongst them running a program (such as Wireshark) that can "capture" network packets and analyze them over the air?

Reviewer: great example! only small question of whether we want to explicitly use the name 'wireshark' and associate with bad actor? not sure if I'm being overly cautious though!

Reviewer 2: Good call out, but yes, Wireshark is a tool common to professionals and hackers alike. It's the most common tool used to monitor internet traffic over the air, capture network packets, and read unencrypted traffic. They're well aware of their status as a tool used for both good and evil.

> A typical scenario that might be seen frequently is an employee working in a public space, such as an airport or café. The user might be connected to an unsecured public network offered by a facility to which many others are also connected. What if there is a bad actor amongst them running a program (such as Wireshark) that can "capture" network packets and analyze them over the air?

Reviewer 2: ```suggestion
A typical scenario that might be seen frequently is an employee working in a public space, such as an airport or café. The user might be connected to an unsecured public network offered by a facility to which many others are also connected. What if there is a bad actor amongst them running a program that can "capture" network packets and analyze them over the air?
```

### Git sync

dbt Cloud can sync with a variety of git providers, including [Github](/docs/cloud/git/connect-github), [Gitlab](/docs/cloud/git/connect-gitlab), and [Azure DevOps](/docs/cloud/git/connect-azure-devops) within its integrated development environment ([IDE](/docs/cloud/develop-in-the-cloud)). Communication takes place over HTTPS rather than SSH and is protected using the TLS 1.2 protocol for data in transit.

The git repo information is stored on dbt Cloud servers to make it accessible during the IDE sessions. When the git sync is disabled, you must [contact support](mailto:support@getdbt.com) to request the deletion of the synced data. 

### Authentication services

The default settings of dbt Cloud enable local users with credentials stored in dbt Cloud. Still, integrations with various authentication services are offered as an alternative, including [single sign-on services](/docs/cloud/manage-access/sso-overview). Access to features can be granted/restricted by role using [RBAC](/docs/cloud/manage-access/enterprise-permissions).

> SSO features are essential because they reduce the number of credentials a user must maintain. Users sign in once and the authentication token is shared among integrated services (such as dbt Cloud). The token expires and must be refreshed at predetermined intervals, requiring the user to go through the authentication process again. If the user is disabled in the SSO provider service, their access to dbt Cloud is disabled, and they can not override with local auth credentials. 

Reviewer: ```suggestion
SSO features are essential because they reduce the number of credentials a user must maintain. Users sign in once and the authentication token is shared among integrated services (such as dbt Cloud). The token expires and must be refreshed at predetermined intervals, requiring the user to go through the authentication process again. If the user is disabled in the SSO provider service, their access to dbt Cloud is disabled, and they cannot override this with local auth credentials. 
```
